AIO Fighter fixes & New QoL functionality

See1Duck opened the pull request:
# Changelog


---

## AIOFighterPlugin
**Improve combat, potion, and loot handling.**  
- Refactored the plugin to streamline potion consumption into a unified system, removing manual toggles for individual potions.  
- Updated loot logic to dynamically determine minimum free inventory slots, enhancing flexibility while looting.  
- Simplified state checks for broader compatibility across banking, walking, and combat scenarios.

**Improve banking logic and walking behavior.**  
- Enhanced the banking process by ensuring the bank interface is closed after transactions for better flow.  
- Updated walking logic to handle edge cases where center location has invalid coordinates (e.g., `[0, 0, 0]`).  
- Improved state transitions by switching to `IDLE` upon successful navigation.  

These changes result in more reliable bot behavior and fewer interruptions during automated activities.

### Highlights
- **Consolidated potion management**: Ensures consistent usage of antidotes, combat potions, and prayer potions without separate config toggles.  
- **Dynamic inventory calculations**: Replaced hardcoded slot checks with a system tied to banking preferences.  
- **New slayer mode configurations**: Added hidden features to prepare for future Slayer-specific functionality.

---

## CraftingManager
**Add quick crafting for dragonhide.**  
- Introduced a "Quick Craft" menu entry when using a needle on leather, automating interactions and reducing manual input.  

## GemCuttingManager
**Add quick gem cutting functionality.**  
- Enabled automated gem cutting through a "Cut Gem" menu entry when using a chisel.  
- Supports level-based gem selection, speeding up crafting operations while ensuring only valid gems are processed.

## UncutGems & DragonhideCrafting
**Add crafting and cutting enum support.**  
- Created enums for uncut gems and dragonhide items, streamlining retrieval of craftable/cutable items based on player skill levels.  
- Improves inventory management logic and overall code organization.

## QoLPlugin
**Extend support for crafting and alchemy features.**  
- Enhanced "Quick High Alch" functionality by adding a menu entry for high-alching profitable items directly from the inventory.  
- Registered both `CraftingManager` and `GemCuttingManager` for a more seamless experience with crafting tasks.

## WintertodtScript
**Improve warmth management and food usage.**  
- Added a function to monitor the player's warmth during Wintertodt.  
- Automatically consumes rejuvenation potions when below a certain threshold to improve survival and efficiency.

---

## AIOFighterPlugin
**Remove unused potion-handling scripts.**  
- Deleted `AntiPoisonScript` and `CombatPotionScript` as they were outdated and no longer in use.  
- Reduced code complexity and redundant logic for better maintainability.

---

## PrayerPotionScript
**Fix unnecessary prayer potion check removal.**  
- Removed the `togglePrayerPotions` configuration check, which caused failures in potion usage logic.  
- Ensures correct handling of prayer potion consumption based solely on prayer level thresholds.

See1Duck committed:
Create nightly_artifact_release.yml

test

See1Duck committed:
Update nightly_artifact_release.yml

test2

See1Duck committed:
Update nightly_artifact_release.yml

test3

See1Duck committed:
Merge branch 'chsami:development' into development

See1Duck committed:
Merge branch 'chsami:development' into development

See1Duck committed:
Merge branch 'chsami:development' into development

See1Duck committed:
- **BankerScript**: Improve banking logic and walking behavior.

  Enhanced the banking process by ensuring the bank interface is closed after transactions for better flow. Updated walking logic to handle edge cases where center location has invalid coordinates (e.g., [0, 0, 0]) and improved state transitions by switching to `IDLE` upon successful navigation. These changes ensure more reliable bot behavior and reduced interruptions during automated activities.

See1Duck committed:
- **AIOFighterPlugin**: Improve combat, potion, and loot handling.

  Refactored the plugin to streamline potion consumption into a unified system, removing manual toggles for individual potions. Updated loot logic to dynamically determine minimum free inventory slots, enhancing flexibility for looting logic during combat. Simplified state checks for better compatibility across various conditions like banking and walking.

  - Consolidated potion management into an automated system, ensuring consistent usage of antidotes, combat potions, and prayer potions regardless of config toggles.
  - Replaced hardcoded inventory slot handling in loot scripts with dynamic calculation based on banking preferences.
  - Introduced new slayer mode configurations (hidden) for integrating future Slayer-specific functionality.

See1Duck committed:
- **CraftingManager**: Add quick crafting for dragonhide.

  Implemented a feature to streamline dragonhide crafting by introducing "Quick Craft" menu entries when using a needle on leather. This update automates interaction sequences and reduces manual input.

- **GemCuttingManager**: Add quick gem cutting functionality.

  Enabled automated gem cutting through a "Cut Gem" menu item when using a chisel, with support for level-based gem selection. This significantly speeds up crafting operations while ensuring only valid gems are processed.

- **UncutGems & DragonhideCrafting**: Add crafting and cutting enum support.

  Added enums to categorize uncut gems and dragonhide items, supporting easy retrieval of craftable/cutable items based on player skill levels. This abstraction improves inventory management logic.

- **QoLPlugin**: Extend support for crafting and alchemy features.

  Improved "Quick High Alch" functionality by adding a menu entry for high-alching profitable items directly from the inventory. Registered `CraftingManager` and `GemCuttingManager` for a seamless user experience when performing crafting tasks.

- **WintertodtScript**: Improve warmth management and food usage.

  Added a function to monitor the player's warmth during Wintertodt and automatically consume rejuvenation potions when below a threshold, enhancing survival during minigame participation.

See1Duck committed:
- **AIOFighterPlugin**: Remove unused potion-handling scripts.

  Deleted `AntiPoisonScript` and `CombatPotionScript`, as their functionality is outdated and no longer utilized by the plugin. This cleanup reduces code complexity and eliminates redundant logic, ensuring better maintainability for future updates.

  - AntiPoison functionality was removed due to lack of relevance in botting scenarios.
  - Combat potion handling was deprecated in favor of integrated, more efficient solutions elsewhere in the plugin.

See1Duck committed:
- **SlayerPlugin**: Add SlayerMaster enum for NPC task assignment.

  Implemented a new `SlayerMaster` enum to define Slayer masters and their respective WorldPoint locations. This addition standardizes access to Slayer master data, improving code readability and enabling more efficient task assignment for automation scripts.

See1Duck committed:
- **PrayerPotionScript**: Fix unnecessary prayer potion check removal.

  Removed an unnecessary configuration check (`togglePrayerPotions`), which caused failure in potion usage logic during combat. This ensures the script correctly handles prayer potion consumption by focusing solely on prayer level thresholds without additional conditions.

See1Duck committed:
Update nightly_artifact_release.yml

removed cron job

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/aiofighter/combat/PotionManagerScript.java:
```diff
@@ -16,44 +16,54 @@ public boolean run(AIOFighterConfig config) {
     }
+
+    public int getPrayerPotionThreshold() {
+        int maxRestore = 7 + (Rs2Player.getRealSkillLevel(Skill.PRAYER) / 4);
```
Prayer potions follow this math for the threshold, but super restores use the following

![image](https://github.com/user-attachments/assets/9625a392-b452-43f3-86a2-a1e0333b9015)

Then Moonlight Potions are the following:

![image](https://github.com/user-attachments/assets/bae6e70b-4bdd-48ff-b065-428d0fe8c073)

Possibly we manage these sort of checks inside of a Rs2Player.drinkPrayerPotion to determine the threshold based on what is in the inventory & then drink the potion using their respective threshold

gmason0 replied:
I did make this as apart of the suggestion that was created yesterday, I am not sure if we want to leverage this or something else you had in mind?

```java
public static boolean drinkPrayerPotion() {
    int maxPrayer = getRealSkillLevel(Skill.PRAYER);
    int maxHerblore = getRealSkillLevel(Skill.HERBLORE);
    int restoreAmount;

    if (hasPotion("moonlight moth mix")) {
        restoreAmount = 22;
    } else if (hasPotion("moonlight potion")) {
        int prayerRestore = (maxPrayer / 4) + 7;
        int herbloreRestore = (int) Math.floor((maxHerblore * 3.0 / 10.0)) + 7;
        restoreAmount = Math.max(prayerRestore, herbloreRestore);
    } else if (hasPotion("super restore") || hasPotion("blighted super restore")) {
        restoreAmount = (maxPrayer / 4) + 8;
    } else {
        restoreAmount = (maxPrayer / 4) + 7;
    }

    int threshold = maxPrayer - restoreAmount;
    int randomizedThreshold = Rs2Random.randomGaussian(threshold - 5, 3);
    randomizedThreshold = Math.min(randomizedThreshold, threshold);
    System.out.println("Threshold: " + randomizedThreshold);

    return drinkPrayerPotionAt(randomizedThreshold);
}
```

See1Duck replied:
Yeah i get the point and it would be better to move it to the main method instead. The only issue i see it that if you have multiple different potions in the inv or if some are noted as a drop resulting in the wrong calc. not sure how we can get around that

See1Duck replied:
Lets go with your approach, ill implement that instead in my pr instead

gmason0 replied:
Sounds good, I also will probably refactor this to be based on item order in the inventory but i think now having priorities is alright as that is how the current  drinkPrayerPotionAt functions

See1Duck committed:
- **AIOFighter**: Refactor prayer potion logic for better customization.

  Replaced hardcoded prayer potion threshold logic with a new method in `Rs2Player`, enabling better potion-specific threshold calculations. This allows for more accurate handling of different potion types (e.g., moonlight potion, super restore) and introduces randomized behavior for improved anti-detection.

- **Rs2Player**: Add dynamic prayer potion restoration logic.

  Introduced `drinkPrayerPotion()` with support for multiple potion types and their respective prayer restoration calculations. The method uses randomized thresholds to mimic human-like behavior, enhancing bot sophistication and reducing detection risks.

See1Duck committed:
Merge remote-tracking branch 'origin/development' into development
